External/diffmatchpatch/DiffMatchPatchCFUtilities.c

```diff
@@ -142,7 +145,11 @@ CFIndex diff_commonSuffix(CFStringRef text1, CFStringRef text2) {
     char2 = CFStringGetCharacterFromInlineBuffer(&text2_inlineBuffer, (text2_length - i));
 
     if (char1 != char2) {
-      return i - 1;
+        if ( CFStringIsSurrogateLowCharacter(char1) || CFStringIsSurrogateHighCharacter(char1) ) {
+            return MIN(i - 2, 0);
+        }
+
+        return i - 1;
```

Comment:
did you see https://github.com/google/diff-match-patch/pull/69#issuecomment-525834911 where new bugs appeared by using this approach?